pkg/oci/oci.go

```diff
@@ -305,6 +306,48 @@ func ListGadgetImages(ctx context.Context) ([]*GadgetImageDesc, error) {
 	return imageColumns, nil
 }
 
+func deleteTree(ctx context.Context, store *oci.Store, root ocispec.Descriptor) error {
+	descriptors, err := content.Successors(ctx, store, root)
+	if err != nil {
+		return fmt.Errorf("getting image successors: %w", err)
+	}
+
+	for _, descriptor := range descriptors {
+		deleteTree(ctx, store, descriptor)
+	}
+
+	predecessors, err := store.Predecessors(ctx, root)
+	if err != nil {
+		return fmt.Errorf("getting image predecessors: %w", err)
+	}
+
+	if len(predecessors) > 1 {
```

Comment:
I am not so sure how `.Predecessors()` work internally. But if it is like in a graph then I think that this check has to be the first thing in the function.

Example (All letters are descriptors here):
my-image:v0.3:     `A->B->C->D`
my-image:v0.2:     `Z->B->C->D`

Predecessors(B): 2
Predecessors(C): 1
Predecessors(D): 1

C and D would be deleted if we call `deleteTree(Z)`

Am I thinking right about this?